### PR TITLE
Refine Image Classification Codes&Docs

### PR DESCRIPTION
**Pull request by shippingwang**
**Doc & Folder Structure**
- [x] refine readme doc, remove the defunct pics, comments, etc.
- [x] refine readme_cn doc, translate args explanation.
- [x] remove models/ to legacy/, rename models_name/ to models/, and revise associated codes in the **train/eval/infer** python files.
- [x] ~~remove PIL reader to legacy/, keep the cv2 reader as default reader.~~
- [x] move utility.py to utils
- [x] Add Stage3 release pretrained models.
---
 **Train.py**:
- [x]  Add args ```optimizer``` , etcs , add comments to explain how to train ImageNet when using [RMSProp](http://paddlepaddle.org/documentation/docs/zh/1.3/api_cn/optimizer_cn.html#rmspropoptimizer)
        For now, we do not add this args.
- [x] polish learning rate calculation, Using "ceil" 
- [x] polish log format.
- [x] print learning rate while traing models.
- [x] use cosine_decay api instead of python implementation.
---
**Reader.py**
- [x] ~~**reader.py**~~ reader_cv2.py add cv2 resize parameters.
---
**Others**
- [x] polish **infer.py**, use "==" to compare instead of "is", etc.
- [x] fix the conflicts causing by redundacy "softmax" activation codes both in the model architecutre and **train/eval/infer** python files.
- [x] Add save_inference args in infer.py
- [x] add cosine_decay_with_warmup to support shufflenetv2 in utility.py
- [x] Update run.sh

### CODE REVIEW

**Comment by shippingwang** [comment]
验证
1. train.py 验证
image

2.eval.py验证
python eval.py --model=AlexNet --pretrained_model=/work/data/cv/pretrain/AlexNet_pretrained/
image

3. infer.py验证
python infer.py --model=AlexNet --pretrained_model=/work/data/cv/pretrain/AlexNet_pretrained/ 
image



4.CV2 reader 正确性，和PIL 对比效果
~~Moblenet V1 20epoch~~
~~PIL:~~
image
~~CV2:~~
image
需进一步验证

File: fluid/PaddleCV/image_classification/train.py [hunk 1]
```diff
@@ -119,6 +104,25 @@ def optimizer_setting(params):
     elif ls["name"] == "adam":
         lr = params["lr"]
         optimizer = fluid.optimizer.Adam(learning_rate=lr)
+    elif ls["name"] == "RMSProp_cosine":
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/train.py]: 都小写 RMSProp_cosine -> rmsprop_cosine_decay

**Comment by shippingwang** [comment]
NOTE：softmax 都放在train 里面

File: fluid/PaddleCV/image_classification/README.md [hunk 2]
```diff
@@ -216,20 +185,12 @@ Models are trained by starting with learning rate ```0.1``` and decaying it by `
...
-
-
-- Released models: not specify parameter names
-
-**NOTE: These are trained by using model_category=models**
-
-|model | top-1/top-5 accuracy(PIL)| top-1/top-5 accuracy(CV2) |
-|- |:-: |:-:|
-|[ResNet152](http://paddle-imagenet-models.bj.bcebos.com/ResNet152_pretrained.zip) | 78.18%/93.93% | 78.11%/94.04% |
-|[SE_ResNeXt50_32x4d](http://paddle-imagenet-models.bj.bcebos.com/se_resnext_50_model.tar) | 78.32%/93.96% | 77.58%/93.73% |
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/README.md:235]: 把这个表格暂且放到legacy里，同时在legacy里声明下，模型配置有没有name，麻烦自己在注意下措辞。

File: fluid/PaddleCV/image_classification/README_cn.md [hunk 3]
```diff
@@ -66,54 +64,49 @@ python train.py \
...
+* **pretrained_model**: 预训练模型路径，默认值: None
+* **checkpoint**: 用于继续训练的检查点（指定具体模型存储路径，如"output/SE_ResNeXt50_32x4d/100/"），默认值: None
+* **fp16**: 是否开启混合精度训练，默认值: False
+* **scale_loss**: 调整混合训练的loss scale值，默认值: 1.0
+* **l2_decay**: l2_decay值，默认值: 1e-4
+* **momentum_rate**: momentum_rate值，默认值: 0.9
+
+在```run.sh```中有用于训练的脚本.
+
+**数据读取器说明：** 数据读取器定义在```reader.py```中。一般, CV2可以提高数据读取速度, PIL reader可以得到相对更高的精度, 我们现在提供基于CV2的数据读取器, 在[训练阶段](#模型训练), 默认采用的增广方式是随机裁剪与水平翻转, 而在[模型评估](#模型评估)与[模型预测](#模型预测)阶段用的默认方式是中心裁剪。当前支持的数据增广方式有：
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/README_cn.md]: > 一般, CV2可以提高数据读取速度, PIL reader可以得到相对更高的精度, 我们现在提供基于CV2的数据读取器

reader部分还原吧，继续默认是PIL reader。 可以说明下，我们默认使用PIL的reader。

上面CV2 reader速度和PIL reader精度结论部分暂且去掉吧。

File: fluid/PaddleCV/image_classification/README_cn.md [hunk 4]
```diff
@@ -66,54 +64,49 @@ python train.py \
...
 * 旋转
 * 颜色抖动
 * 随机裁剪
 * 中心裁剪
 * 长宽调整
 * 水平翻转
 
-**训练曲线：** 通过训练过程中的日志可以画出训练曲线。举个例子，训练AlexNet出来的日志如下所示：
+**训练结果：** AlexNet的训练日志如下所示：
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/README_cn.md]: 下面log也可以去掉，我们log经过改动，已不是这样了。

File: fluid/PaddleCV/image_classification/README_cn.md [hunk 5]
```diff
@@ -152,15 +145,11 @@ Testbatch 20,loss 1.32524681091, acc1 0.6875,acc5 0.9375,time 0.37 sec
 Testbatch 30,loss 1.46830511093, acc1 0.5,acc5 0.9375,time 0.51 sec
 Testbatch 40,loss 1.12802267075, acc1 0.625,acc5 0.9375,time 0.35 sec
 Testbatch 50,loss 0.881597697735, acc1 0.8125,acc5 1.0,time 0.32 sec
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/README_cn.md]: 同上，log也可以去掉。

**Review by qingqing01 — APPROVED** [review]
